speculos/client.py

```diff
@@ -0,0 +1,181 @@
+        stream = self.session.get(f"{self.api_url}/events?stream=true", stream=True)
+        assert stream.status_code == 200
+        return stream
+
+    def get_next_event(self):
+        line = self.stream.raw.readline()
+        event = json.loads(line)
+        assert event
+        return event
+
+    def wait_for_text_event(self, text: str) -> dict:
+        """Wait until an event containing the specified text is received."""
+
+        while True:
+            event = self.get_next_event()
+            if text in event["text"]:
+                break
+        return event
+
+    def press_and_release(self, button: str):
+        assert button in ["left", "right", "both"]
+        data = {"action": "press-and-release"}
+        with self.session.post(f"{self.api_url}/button/{button}", json=data) as response:
+            assert response.status_code == 200
+
+    def finger_touch(self, x: int, y: int):
+        data = {"action": "press-and-release", "x": x, "y": y}
+        with self.session.post(f"{self.api_url}/finger", json=data) as response:
+            assert response.status_code == 200
+
+    def get_screenshot(self) -> bytes:
+        with self.session.get(f"{self.api_url}/screenshot") as response:
+            assert response.status_code == 200
+            assert response.headers["Content-Type"] == "image/png"
+            return response.content
+
+    def _apdu_exchange(self, data: bytes) -> bytes:
+        with self.session.post(f"{self.api_url}/apdu", json={"data": data.hex()}) as response:
+            apdu_response = ApduResponse(response)
+            return apdu_response.receive()
+
+    def _apdu_exchange_async(self, data: bytes) -> requests.Response:
+        return self.session.post(f"{self.api_url}/apdu", json={"data": data.hex()}, stream=True)
+
+    def set_automation_rules(self, rules: dict):
+        with self.session.post(f"{self.api_url}/automation", json=rules) as response:
+            assert response.status_code == 200
+
+
+class SpeculosInstance:
+    def __init__(self, app: str, args=[]) -> None:
+        self.app = app
+        self.args = args
+        self.process = None
+
+        if "--display" not in self.args:
+            self.args += ["--display", "headless"]
+
+        if "--api-port" not in self.args:
+            self.port = 5000
+        else:
+            n = self.args.index("--api-port")
+            self.port = int(self.args[n])
+
+    def _wait_until_ready(self):
+        connected = False
+        for i in range(0, 20):
+            try:
+                s = socket.create_connection(("127.0.0.1", self.port))
+                connected = True
+                break
+            except ConnectionRefusedError:
+                time.sleep(0.1)
+
+        if not connected:
+            raise f"Failed to connect to the speculos instance on port {self.port}"
+
+        s.close()
+
+    def start(self):
+        cmd = ["python3", "-m", "speculos.main"] + self.args + [self.app]
```

Comment:
`"python3"` should be replaced with `sys.executable` (https://docs.python.org/2.7/library/sys.html#sys.executable)